### PR TITLE
Update ml-image environment conda lock process

### PR DESCRIPTION
**Pull request by rabernat**
As discussed in #260, this PR changes the ml-image so that it extends the pangeo-notebook image.

I verified that the conda-lock command works locally.

### CODE REVIEW

File: ml-notebook/environment.yml [hunk 1]
```diff
@@ -1,39 +1,11 @@
...
 channels:
  - conda-forge
  - defaults
 dependencies:
- - python=3.9*
- - pangeo-notebook=2021.09.30
- - pip
  - conda-forge/linux-64::cupy
  # Using defaults channel for GPU support https://github.com/pangeo-data/pangeo-docker-images/issues/243
  - pkgs/main/linux-64::tensorflow-gpu>=2
```
[the remark this answers is not in the document]
Comment (rabernat) [reply ml-notebook/environment.yml:10]: Yes GPU support is very important.

**Comment by rabernat** [comment]
It's frustrating that this change ballooned the image size so much. I'm not sure what to do.

**Comment by sgibson91** [comment]
@rabernat Are you using multi-stage builds? This is something I've been meaning to investigate for smaller image sizes. I have [this article](https://devopsdirective.com/posts/2021/04/tiny-container-image/) saved in my "To Read" database that is a (quite extreme!) example of this process but may be helpful here.

**Comment by scottyhq** [comment]
> using multi-stage builds? 
not currently. could be interesting to try https://github.com/mamba-org/micromamba-docker as well, but there is something reassuring about just using ubuntu :) 

> It's frustrating that this change ballooned the image size so much
I expect this image to be around pangeo-notebook+1GB anyways since these ML packages add up quickly:
```
    cudatoolkit-10.1.243       |       h6bb024c_0       347.4 MB
    cudnn-7.6.5                |       cuda10.1_0       179.9 MB
    cupti-10.1.168             |                0         1.4 MB
    cupy-8.3.0                 |   py39hcaf9a05_0        22.7 MB
    mkl-2021.3.0               |     h06a4308_520       141.2 MB
    tensorflow-base-2.4.1      |gpu_py39h29c2da4_0       195.2 MB
```

I suppose we can just merge and try to sort out smaller images later.

**Review by scottyhq — APPROVED** [review]

**Pull request merged**